include/enum_fortran.inc

```diff
@@ -67,3 +67,16 @@ enum, bind(c)
   enumerator :: meta_uint64  = -1
   enumerator :: meta_string  = 7
 end enum
+
```

Comment:
The use of explicitly defined enums is necessary to make sure that all the languages are communicating the same thing. For the Fortran user though, I wonder if it's worth writing an extra subroutine that explicitly says what the error is, e.g.

```
function SR_error_parser( code ) result( error )
  integer(kind=enum_kind), intent(in) :: code
  logical, intent(  out) :: error
  
  error = .true.
  select case (code)
    case(0):
      error = .false.
    case(1):
      write(stderr,*) "Bad allocation"
    case default:
       write(stderr,*) "Invalid Error"
      
end function SR_error_parser
```

Something simple for now would work, later on it might be worth making a derived type (and extensions) to make this more extensible.

Reply:
Done